scripts/utilities/cellspace.py

```diff
@@ -537,65 +652,140 @@ def add_height_map(self, top_left, height_map, temp_mod=False):
         x_len, y_len, = height_map.shape
         #image_window = self.cells[x_min:x_min + x, y_min:y_min + y]
 
-        sm = self.surface_map()
+        sm = self.surface_map
         for x in range(x_len):
             for y in range(y_len):
                 h = sm[x_min + x, y_min + y]
                 for i in range(h-height_map[x,y]+1, h+1):
                     if temp_mod:
-                        self.temp_mod_cells[x_min+x,i,y_min+y] = CellSpace.cell_types['grain']
+                        self.temp_mod_cells[x_min+x,i,y_min+y] = cell_types['grain']
                     else:
-                        self.cells[x_min+x,i,y_min+y] = CellSpace.cell_types['grain']
+                        self.cells[x_min+x,i,y_min+y] = cell_types['grain']
 
     # add a sin wave to the cellspace
-    def add_sine(self, amp=5, f=10):
+    def add_sine(self, amp=5, f=10, temp_mod=False):
         # mgrid of (l,d) values
         a = np.mgrid[f*-1.0:f*1.0:f*2.0/self.length, f*-1.0:f*1.0:f*2.0/self.depth]
```

Comment:
Is there a way to make this math clearer? I don't know what it's doing.

Reply:
Or comment to explain logic, if possible

Reply:
This has been updated to use an inverse Fourier transform to create the sinusoidal height map. The amplitude corresponds directly to the height in the height maps. The frequency is now relative to the size of the height map in the horizontal or vertical directions.